lti_provider/validator.py

```diff
@@ -23,46 +25,55 @@ def enforce_ssl(self):
             ssl = True
         return ssl
 
-    def check_client_key(self, client_key):
-        """
-        Check client key is provided correctly and LII Consumer with that key exists
-
-        :param client_key: client key from LTI request
-        :return: boolean flag
-        """
-        logger.debug('Client key is checking')
-        try:
-            self.consumer = Consumer.objects.get(consumer_key=client_key)
-        except Consumer.DoesNotExist:
-            raise oauth1.OAuth1Error('Consumer with the key {} is not found.'.format(client_key))
-        return super(RequestValidator, self).check_client_key(client_key)
-
     def validate_timestamp_and_nonce(self, client_key, timestamp, nonce, request):
         """
-        Validate LTI Consumer has not expired key
+        Validate LTI request's timestamp and nonce
 
         :param client_key: client key from LTI request
         :param timestamp: timestamp from LTI request
         :param nonce: nonce from LTI request
         :param request: LTI request
         :return: boolean flag
         """
+        message = "LTI request's {} is not valid."
+
         logger.debug('Timestamp validating is started.')
-        today = datetime.date.today()
-        consumer_expired_date = self.consumer.expiration_date
-        if consumer_expired_date and consumer_expired_date < today:
-            raise oauth1.OAuth1Error('Consumer Key is expired.')
+        timestamp = int(timestamp)
+        timestamp_key = '{}_timestamp'.format(client_key)
+        cache_timestamp = self.cache.get(timestamp_key, timestamp)
```

Comment:
Timestamp is already checked that it is within `timestamp_lifetime` from the current time in `oauthlib.oauth1.rfc5849.endpoints.base` L167. Here we validate timestamp where validation is: 
```
The timestamp value MUST be a positive integer and MUST be equal or greater than the timestamp used in previous requests.
```
Taken from the documentation [link](https://oauth.net/core/1.0/#nonce)